var/spack/repos/builtin/packages/gtkplus/package.py

```diff
@@ -34,10 +34,14 @@ class Gtkplus(Package):
         '2.24.25', '612350704dd3aacb95355a4981930c6f',
         url="http://ftp.gnome.org/pub/gnome/sources/gtk+/2.24/gtk+-2.24.25.tar.xz")
 
+    variant('X', default=True, description="Enable an X toolkit")
+
     depends_on("atk")
     depends_on("gdk-pixbuf")
-    depends_on("pango")
     depends_on("glib")
+    depends_on("pango")
```

Comment:
Unfortunately, it isn't redundant. Try running something like:
```
$ spack spec gtkplus ^pango@1.2.3
```
If you don't have `depends_on('pango')`, Spack will tell you that `gtkplus` does not depend on `pango`. If you run:
```
$ spack spec gtkplus +X ^pango@1.2.3
$ spack spec gtkplus -X ^pango@1.2.3
```
then it will work. This problem is most common in packages that `depends_on('hdf5+mpi', when='+mpi')`, for example. This is a long-standing bug in the concretizer. It doesn't evaluate variants during normalization unless they are explicitly asked for. Therefore, it doesn't know that it depends on `pango`. @tgamblin can give you more details.

In my opinion, the best solution would be variant forwarding. These 3 lines are kind of redundant. We should probably fix this bug anyway.